go/private/go_repository.bzl

```diff
@@ -57,6 +57,13 @@ def _new_go_repository_impl(ctx):
   if result.return_code:
     fail("failed to generate BUILD files for %s: %s" % (
         ctx.attr.importpath, result.stderr))
+  ctx.file('WORKSPACE', """
+# Automatically generated by //go/private:go_repository.bzl
+workspace(name = "{workspace_name}")
+{repo_deps}
+""".format(
+    workspace_name = ctx.name,
+    repo_deps="\n".join(["local_repository(name='%s', path='/dummy')" % r for r in ctx.attr.repo_deps])))
```

Comment:
Using `local_repository` here doesn't make sense to me. Shouldn't Bazel report a conflict if you declare a `local_repository` in one place and, for example, a `git_repository` in another place? Seems like that should count as different versions.

Maybe there should be a special `use_repository` rule?